dimod/binary/binary_quadratic_model.py

```diff
@@ -840,8 +854,13 @@ def add_quadratic_from_dense(self, quadratic: ArrayLike):
         """Add quadratic biases from a square 2d array-like.
 
         Args:
-            quadratic:
-                An square 2d `array_like`_ of quadratic biases.
+            quadratic: Quadratic biases as a square 2d `array_like`_.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel("BINARY")
```

Comment:
IMO we should also include a SPIN model to show how the diagonal is handled.

Should also print the whole thing.

Reply:
@arcondello, have you tested the handling of diagonals for SPIN models? I don't understand the relation between the `pybqm.py` and `cybqm/cybqm_template.pyx.pxi` files but when you set a non-zero diagonal it triggers an exception from line https://github.com/dwavesystems/dimod/blob/3bf1ab194e6875cb7f4ac99ef75d4985f3360a4b/dimod/binary/cybqm/cybqm_template.pyx.pxi#L282 :

```
bqm.add_quadratic_from_dense([[0.2, -0.4],[0.1, -0.2]])

dimod\binary\cybqm\cybqm_template.pyx.pxi in dimod.binary.cybqm.cybqm_float64.cyBQM_template.add_quadratic_from_dense()

ValueError: 0 cannot have an interaction with itself
```

Reply:
Ah, I forgot what the exact behaviour is. But in this case we should mention that values on the diagonal raise an exception.

Reply:
Will add a `Raises`